File: mario/mario.c
#include <cs50.h>
#include <stdio.h>

int get_height(void);

int main(void)
{
    //print pyramide
    int height = get_height();
    int spaces = height - 1;

    for (int i = 0; i < height; i++)
    {
        for (int j = 0; j < spaces; j++)
        {
            //print spaces
            printf(" ");
        }
        int count = 2;
        while (count)
        {
            //print hashes 2 times with spaces separator
            for (int k = 0; k < height - spaces; k++)
            {
                //print hashes on row
                printf("#");
            }
            if (count == 2)
            {
                printf("  ");
            }
            count--;
        }
        printf("\n");
        spaces--;
    }
}

int get_height(void)
{
    //get height from user
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);
    return height;
}

File: sentimental-mario-more/mario.py
# print out a half-pyramid of a specified height
def main():
    height = get_height()
    print_pyramid(height)


# prompt the user for the half-pyramid’s height, a positive integer between 1 and 8, inclusive.
def get_height():
    while True:
        try:
            height = int(input("Enter the height: "))
            if (height < 1 or height > 8):
                continue
            return height
        except ValueError:
            continue


# generate half-pyramid of a specified height
def print_pyramid(height):
    spaces_count = height - 1
    hashes_count = 1

    while spaces_count >= 0:
        print_spaces(spaces_count)
        print_hashes(hashes_count)
        print("  ", end="")
        print_hashes(hashes_count)
        print("")
        spaces_count -= 1
        hashes_count += 1


# print spaces
def print_spaces(count):
    if count > 0:
        for i in range(count):
            print(" ", end="")


# print hashes
def print_hashes(count):
    for i in range(count):
        print("#", end="")


if __name__ == "__main__":
    main()
